### DISCUSSION TITLE
Repository for save states to assist with future tickets

### DISCUSSION THREAD

**Discussion by suspect15** [RetroAchievements/RAWeb#1095] [Ideas]
I'd be surprised if this hasn't been suggested before, but I couldn't find anything from my searching so I'll throw it out anyway.

I've been browsing through open tickets seeing which ones I thought I might be able to resolve and the lack of available saves makes many of the tickets quite difficult to address.  I'm very much an inexperienced dev, but I do keep a ton of save states from the games I've worked on and was thinking it might not be a bad resource to have a repository that devs could upload whatever save states they used during set creation.  It would make future tickets significantly easier to address when devs become inactive.

If this is idea is supported, I'd strongly encourage making it part of set requirements moving forward as I don't imagine it would be a significant burden.  I'm not sure if github or on RA itself is the best place to store these saves.  Someone smarter than me should weigh in on that if this idea is deemed reasonable.

I understand SnowPin has collected a bunch and I'm sure between active devs there we'd have a decent start.

**Comment by televandalist** [comment]
I just messaged SnowPin about this and offered to help get the ball rolling on the repository.

**Comment by Bryan1150** [comment]
The only problem I see is how to maintain it private so those saves are just used with a devving purpose in mind

**Reply by suspect15** [reply]
This is definitely a concern.  One idea I have is if uploading is possible via the RA site under the Dev section:
![image](https://user-images.githubusercontent.com/106546659/185382970-fd96d6e9-e774-4382-8075-b26d604392ba.png)
Then the saves could be stored on site, but access to the saves could be restricted so that a dev would have to get approval to access either a particular game's saves or all saves in general.  I think this would increase the perception of accountability and help ensure the saves were accessed for their intended purpose.

There has to be some level of trust when risk is involved, so finding what that sweet spot is will certainly take some discussion and consideration from the team.

**Reply by TeddyWestside22** [reply]
If it's under the dev section and there is auditing, then it should make it easy to detect any suspicious activity. I like the idea of uploading the saves, as it's a real problem trying to fix older games when it's at a section half-way through the game. I don't know if space is a concern though... for the earlier consoles it's fine, but PSP save states are over 40MB each.

**Reply by meleu** [reply]
> I'd be surprised if this hasn't been suggested before

In the Jurassic days of RA this was the repo for save states: https://github.com/meleu/RetroArch-problematic-cheevos

It was very specific to solve RetroArch incompatibilities though.


> The only problem I see is how to maintain it private so those saves are just used with a devving purpose in mind

A curiosity from a person who is not currently active anymore: what would be the problems of having the save states publicly available? Softcore cheating?

**Reply by Maximdraco** [reply]
Probably cheating in general, is a save state, so, depending of the game, you could load the save state, do a normal save and change to hardcore again, softcore doesn't need to do the last step, only continue playing after loading the save state.

**Reply by meleu** [reply]
This topic makes me remember two things we discussed a lot in the past...

The first is in the Achievement Hunting Rules, the [Not Allowed #3](https://docs.retroachievements.org/Global-Leaderboard-and-Achievement-Hunting-Rules/#3-do-not-use-save-files-or-passwords-that-are-not-your-own-or-were-made-under-other-conditions-of-play):

> Do not use save files or passwords that are not your own or were made under other conditions of play.

I understand that just because it's written it doesn't mean cheaters won't do it. I also understand that it puts a burden on the staff when they have to be investigating cheaters. And this scenario brings the second thing...

In [the RA's mission statement](https://docs.retroachievements.org/Users-Code-of-Conduct/#mission-statement):

> The RetroAchievements project aims to provide tools for creating achievements for retrogames and ways to enjoy such creations.

The reason why "**provide tools for creating achievements**" comes first is that this is RetroAchievements' priority.

It comes first so when we are in a trade-off situation, achievement creation should have more weight in the decision.

Summing up:

Achievement creators are the heroes of this project/community. We should make their job as easy as possible. Creating hindrances for them just because of some cheaters could be violating the "Achievement Hunting Rules" is sad.

**Reply by Jamiras** [reply]
> I don't know if space is a concern though... for the earlier consoles it's fine, but PSP save states are over 40MB each.

This was also brought up in #712.

For most systems where states are so huge, there's also in-game saves which are much smaller (a few hundred KB or less for PSP). They may not put you right at the achievement, but even having something close to the achievement is preferable to having nothing at all. States would be more important for the earlier systems that had no form of in-game save. Unfortunately, there's some in-between systems (like N64) that still have sizable states (16MB) and no in-game saves.

I also prefer saves over states as sometimes bugs are caused simply by loading an in-game save, or by slightly different timings between getting from the save point to the achievement. States created during development are usually just one long session where the dev keeps continuing by loading their save state instead of using the in-game save/password system. Furthermore, states created by the developer  _as they're writing the achievement_ are particularly subject to always working as they represent the exact state of the game that the achievement was designed for.

**Reply by Maximdraco** [reply]
I support this idea, it will help all devs in the site, it could be only accesible to devs and have a message each time a dev enter, or maybe only give acces to devs who had maid a claim for the game, either a normal, revision or collab claim.
